server.js

```diff
@@ -188,9 +186,56 @@ app.get('/:username/:plantId', async (req, res) => {
 });
 
 
+// Edit plant profile
+app.patch('/:username/:plantId/edit', authenticateUser);
+app.patch('/:username/:plantId/edit', async (req, res) => {
+  try {
+    const accessToken = req.header('Authorization');
+    const user = await User.findOne({ accessToken: accessToken });
+    const plantId = req.params.plantId;
+    const { plantname, species, imageUrl, birthday } = req.body;
+    // Find the plant by ID and make sure it belongs to the user
+    const plant = await Plant.findOne({ _id: plantId, user: user._id });
+
+    if (plant) {
+      if (plantname) {
+        plant.plantname = plantname; // Change plantname
+      }
+      if (species) {
+        plant.species = species; // Change species
+      }
+      if (imageUrl) {
+        plant.imageUrl = imageUrl; // Change plant photo
+      }
+      if (birthday) {
+        plant.birthday = birthday; // Change birthday
+      }
+
```

Comment:
does this work if you're only changing one thing? `platname` for example?

Reply:
Yes, as long as it's a .patch-endpoint and not .put you can change only one thing.